src/App.js

```diff
@@ -1,25 +1,43 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+ import axios from 'axios';
+ import { useEffect, useState } from 'react';
+ import Navbar from "./components/Navbar";
+import CountriesList from "./components/CountriesList";
+import CountryDetails from "./components/CountryDetails";
+import {  Route, Routes } from 'react-router-dom';
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+
+  const [countriesArr, setCountriesArr] = useState([]);
+
+
+
+   useEffect(() => {
+    axios.get("https://ih-countries-api.herokuapp.com/countries")
+       .then(response => {
+        setCountriesArr(response.data);
+     })
+       .catch(e => console.log("error getting countries from API", e))
+   }, [])
+
+
+  return <div className="App">    <Navbar />
+  <div className="container">
+    <div className="row">
+      <div className="col-4" style={{maxHeight: '90vh', overflow: 'scroll'}}>
+        <CountriesList countriesArr={countriesArr} />
+      </div>
+      <div className="col-6">
+        <CountryDetails countriesArr={countriesArr}/>
+        <div>
+          <Routes>
+         
+            <Route path="/:alpha3Code" element={ <CountryDetails countriesArr={countriesArr} />} />
+          </Routes> 
+        </div>
+      </div>
```

Comment:
Currently you are rendering CountryDetails twice. You should have only once and inside the Route